### PR TITLE
Add configurable import paths, protobuf enum renames.

### PR DESCRIPTION
**Pull request by robshakir**
* Add support for flags to specify the import paths for `yext.proto` and `ywrapper.proto`.
* Update enumerated name output formats to comply with protobuf best practice.

### CODE REVIEW

File: ygen/genstate.go [hunk 1]
```diff
@@ -108,15 +108,15 @@ func newGenState() *genState {
 // this function in common with enumerations that are not within a union. The name of the enumerated
 // value is calculated based on the original context, and whether path compression is enabled based
 // on the compressPaths boolean.
-func (s *genState) enumeratedUnionEntry(e *yang.Entry, compressPaths bool) ([]*yangEnum, error) {
+func (s *genState) enumeratedUnionEntry(e *yang.Entry, compressPaths, omitUnderscores bool) ([]*yangEnum, error) {
```
Comment (ostromart) [inline ygen/genstate.go]: Some explanation of the reason behind omitUnderscores would be useful. For example why not have the positive version "underscores" and omit all the negation? 
If sticking with negation perhaps change to noUnderscores?

File: ygen/protogen.go [hunk 2]
```diff
@@ -500,6 +502,24 @@ func writeProtoEnums(enums map[string]*yangEnum) ([]string, []error) {
...
+					pp := []string{}
+					for _, pe := range t {
+						pp = append(pp, safeProtoIdentifierName(yang.CamelCase(pe)))
+					}
+					p.ValuePrefix = strings.Join(pp, "_")
+				}
+			}
+
+			if p.ValuePrefix == "" {
+				p.ValuePrefix = enum.name
```
Comment (ostromart) [inline ygen/protogen.go]: Seems like you could just set this by default above the check for Annotation.

File: ygen/genstate.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (robshakir) [reply ygen/genstate.go]: Done.

Comment (robshakir) [reply ygen/genstate.go]: Added a comment re: languages that don't allow this in the styles. I prefer the names with underscores in, hence the preference for this to be on.

File: ygen/genstate.go [hunk 3]
```diff
@@ -233,7 +234,7 @@ func (s *genState) buildDirectoryDefinitions(entries map[string]*yang.Entry, com
 // code generation is required for each enum. Particularly, it removes
 // duplication between config and state containers when compressPaths is true.
 // It also de-dups references to the same identity base, and type definitions.
-func (s *genState) findEnumSet(entries map[string]*yang.Entry, compressPaths bool) (map[string]*yangEnum, []error) {
+func (s *genState) findEnumSet(entries map[string]*yang.Entry, compressPaths, omitUnderscores bool) (map[string]*yangEnum, []error) {
```
[the remark this answers is not in the document]
Comment (robshakir) [reply ygen/genstate.go]: Done.

File: ygen/genstate.go [hunk 4]
```diff
@@ -362,16 +363,18 @@ func (s *genState) findEnumSet(entries map[string]*yang.Entry, compressPaths boo
...
 }
 
 // identityrefBaseTypeFromIdentity takes an input yang.Identity pointer and
 // determines the name of the identity used within the generated code for it. The value
 // returned is based on the defining module followed by the CamelCase-ified version
-// of the identity's name.
-func (s *genState) identityrefBaseTypeFromIdentity(i *yang.Identity) string {
+// of the identity's name. If omitUnderscores is set to false, the module name is
+// separated with undescores, otherwise they are omitted.
+func (s *genState) identityrefBaseTypeFromIdentity(i *yang.Identity, omitUnderscores bool) string {
```
[the remark this answers is not in the document]
Comment (robshakir) [reply ygen/genstate.go]: Done.

File: ygen/genstate.go [hunk 5]
```diff
@@ -381,7 +384,12 @@ func (s *genState) identityrefBaseTypeFromIdentity(i *yang.Identity) string {
 	if definedName, ok := s.uniqueIdentityNames[identityKey]; ok {
 		return definedName
 	}
-	name := fmt.Sprintf("%s_%s", yang.CamelCase(definingModName), yang.CamelCase(i.Name))
+	var name string
+	if !omitUnderscores {
+		name = fmt.Sprintf("%s_%s", yang.CamelCase(definingModName), yang.CamelCase(i.Name))
```
[the remark this answers is not in the document]
Comment (robshakir) [reply ygen/genstate.go]: Done.

**Comment by robshakir** [comment]
Disabling coveralls based on the bug above :-(

File: ygen/protogen.go (hunk 2 above)
Comment (robshakir) [reply ygen/protogen.go]: Done.

File: ygen/genstate.go [hunk 6]
```diff
@@ -470,7 +488,11 @@ func (s *genState) resolveTypedefEnumeratedName(e *yang.Entry) (string, error) {
 		switch len(enumTypes) {
 		case 1:
 			// We specifically say that this is an enumeration within the leaf.
-			typeName = fmt.Sprintf("%s_Enum", enumTypes[0].Name)
+			if !noUnderscores {
```
Comment (ostromart) [inline ygen/genstate.go]: swap to avoid unnecessary negation.

Comment (robshakir) [reply ygen/genstate.go]: Done, thanks.

**Pull request merged**